### PR TITLE
More single-patch support of PR description

### PR DESCRIPTION
**Pull request by webstech**
This is a three part change for PatchSeries.

1. Allow an empty PR description to be passed in (assuming
   this has been validated elsewhere)
2. Handle `footers` present in the PR description with no
   other text (ie no description)
3. Recognize the Based-on: footer.  The case statement was
    missing `:`.

This fixes https://github.com/gitgitgadget/gitgitgadget/issues/220

### CODE REVIEW

File: lib/patch-series.ts [hunk 1]
```diff
@@ -288,7 +285,13 @@ export class PatchSeries {
         let coverLetterBody = prBody.trim();
 
         // parse the footers of the pullRequestDescription
-        const match = prBody.match(/^([^]+)\n\n([^]+)$/);
+        let match = prBody.match(/^([^]+)\n\n([^]+)$/);
+
+        if (!match && !prBody.match(/\n\n/)) {
+            // handle PR descriptions that have no body, just footers
+            match = prBody.match(/^()([A-Za-z]+: [^]+)$/);
```
Comment (dscho) [inline lib/patch-series.ts:292]: I think we need to allow dashes in footer names, like `Based-On:`. I.e. 
```suggestion
            match = prBody.match(/^()([-A-Za-z]+: [^]+)$/);
```

Comment (webstech) [reply lib/patch-series.ts]: Updated - sorry about that.  Added a test for based-on and it was not detected.  Updated the code to detect it.

However, this means based-on may not be well tested in the rest of the code.  I plan to add tests for PatchSeries, which will also test ProjectOptions but may add a separate ProjectOptions test.

Comment (dscho) [reply lib/patch-series.ts]: Thank you!

**Comment by dscho** [comment]
Awesome work, as always!

**Pull request merged**